### ISSUE TITLE
Ionic 3 is creating new instances of providers instead of single shared instance

### ISSUE THREAD

**Issue by donaldruby** [driftyco/ionic-app-scripts#919]
#### Short description of the problem:
Ionic 3 is creating new instances of providers instead of single shared instance.

#### What behavior are you expecting?
Expected behavior is a single shared provider is created. In Ionic 2, things are working as expected. Service is singleton.  But in Ionic 3, which uses angular 4, it looks like every component is creating new instance of that class.

### How to recreate:

1. Define service as class :
```
@Injectable()
export class MyService {
 ...
}
```

2. In other components or pages, import that class with just import command.
```
import { MyService } from '../pages/services/myservice';
```
In constructor:
```
constructor(public _MyService: MyService)
```

3. In main app.module.ts,  add that class service as provider:
```
providers: [someOtherThings, MyService, someOtherThings]
```

**Which @ionic/app-scripts version are you using?**
3.0.1

**Comment by g1bbzy** [comment]
I am having the same problem to want to create new instance of my service every time a page is navigated to....

**Comment by g1bbzy** [comment]
I just used tried the above, within each page constructor; created a public instance of the service. this did not create a new instance of  the service because it still had data on a property created on the apps initial start up.

**Comment by Ovilia** [comment]
I'm having the same problem here. @donaldruby Is this a bug?

**Comment by donaldruby** [comment]
I think it might be related to providers being in a different module. But I'm not sure. I worked around the problem for now by using a non-provider singleton class.

Thanks,
Don

**Comment by Ovilia** [comment]
Sorry, actually I want to ask @danbucholtz if this is a bug, since I saw him assigned himself on this issue. Any progress here?

**Comment by theo4u** [comment]
any update urgent

**Comment by Ovilia** [comment]
For my case, it is cased by `const injector = ReflectiveInjector.resolveAndCreate([AService]);` creating a new instance of `AService`. See detail here: https://forum.ionicframework.com/t/ionic-service-as-provider-is-creating-multiple-instances/90498/8?u=xianzheme

**Comment by theo4u** [comment]
@Ovilia i don't understand, have you solved the issue

**Comment by Ovilia** [comment]
@theo4u Yes, by not using `ReflectiveInjector.resolveAndCreate`. Check if you have it or providers defined other than in `app.module.ts`.

**Comment by theo4u** [comment]
@Ovilia just in the app.module.ts (providers:[]), and still not working

**Comment by Perreiz** [comment]
@theo4u same issue.. did you solve it ?

**Comment by theo4u** [comment]
Yes, I used another method which seems to be cooler all around @perreiz

**Comment by jackletterman** [comment]
Same Issue here, cannot get singleton service to work like referenced in documentation

**Comment by redsharpbyte** [comment]
Hi everyone,

Today I use ionic version 3.7.1 (client 3.19.1) and the last tags on git for ionic are 3.9.x

I could not reproduce the issue described by @donaldruby noticed not-by-all in version 3.0.1

@danbucholtz or @jgw96
You may check some more times, but IMHO, I think you could consider closing this issue ?

**Comment by fairdrop** [comment]
I know this is over 2 years old, but I'm having the same problem still on ionic version 4.12.0

**Comment by donaldruby** [comment]
`import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class MyService {`
